checkQC/parsers/interop_parser.py

```diff
@@ -32,46 +56,135 @@ def get_non_index_reads(summary):
+        the read numbers from.
         :returns: all reads which are not index reads
         """
         non_index_reads = []
         for read_nbr in range(summary.size()):
             if not summary.at(read_nbr).read().is_index():
                 non_index_reads.append(read_nbr)
         return non_index_reads
+    
+    @staticmethod
+    def get_percent_q30_per_cycle(q_metrics, 
+                                  lane_nr, 
+                                  read_nr, 
+                                  is_index_read):
+        """
+        For each lane and read get the mean percent_q30 over swath and tile
+        for each cycle. The first 90% of the ccycles in a read will 
+        initially be included to correct for known biases in the end 
+        of the read, i.e. for a run with 151 cycles, only use the first 
+        136 cycles. Also the first 5 cycles of a read will be removed since 
+        there might be a quality drop there. Lane and read number are given as
+        input and should be 0-indexed, i.e lane_nr should be an integer 0-7
+        and read_nr should be an integer 0-3. The function will return a 
+        dictionary where each key:value pair corresponds to the cycle 
+        nr within the read and the mean '%>= Q30' for all tiles
+        for that given cycle, read and lane.
+        E.g. {6: 98.76343, 7: 98.718155, 8: 98.529205, ...,136: 98.43606}
+        #TODO: Investigate if the quality drop in the beginning of a read
+        is instrument specific. 
+        :param q_metrics: A rectangular array containing values of the 
+        following 'Lane', 'Tile', 'Cycle', 'Read', 'Cycle Within Read', 
+        '%>= Q20', '%>= Q30', 'Surface', 'Swath', and 'Tile Number'.
+        :param lane_nr: int 
+        :param read_nr: int 
+        :param is_index_read: boolean 
+        :returns: {int: float}
+        """
+
+        q30_cycle_df = pd.DataFrame(q_metrics)
+        q30_lane_read = q30_cycle_df[
+                (q30_cycle_df["Lane"] == lane_nr+1) &\
+                (q30_cycle_df["Read"] == read_nr+1)]
+
+        q30_per_cycle = {}
+        if is_index_read:
+            end_cycle = int(max(q30_lane_read["Cycle Within Read"]))
+            start_cycle = end_cycle - 5
+            #Remove the first 2 bases in the index read.
+
+        else:
+            end_cycle = math.ceil(
+                max(q30_lane_read["Cycle Within Read"])*0.9
+            )
+
+            #Remove the last 90% of all cycles since 
+            #they are expected to drop in q30
+            start_cycle = 6
+            #Remove first 5 cycles since they are expected
+            #to have a lower q30
+
+        for cycle in range(start_cycle, end_cycle+1):
+            q30_cycle_mean = float(numpy.mean(
+                q30_lane_read[
+                    (q30_lane_read["Cycle Within Read"] == cycle)
+                ]["%>= Q30"]
+            ))
+            q30_per_cycle[cycle] = q30_cycle_mean
+
+        return q30_per_cycle
```

Comment:
```suggestion
        return {
            cycle: q30_lane_read[
                (q30_lane_read["Cycle Within Read"] == cycle)
            ]["%>= Q30"].mean()
            for cycle in range(start_cycle, end_cycle+1)
        }
```
And remove dictionary initialization on line 103

Reply:
I remember we had a discussion about this during our pair-programming session. For me, your suggested changes are harder for me to read. However, if you have strong opinions on this I can go ahead and change it :slightly_smiling_face:

Reply:
This is the idiomatic way of constructing a dictionary in Python.

Also, more than just turning a loop into a dict-comprehension, it addresses the following points:
- it uses the dataframe's `.mean()` instead of numpy's , and solve the float issue we have
- it removes intermediate temporary variables